pilot/pkg/networking/core/cluster_builder.go

```diff
@@ -139,6 +143,18 @@ func NewClusterBuilder(proxy *model.Proxy, req *model.PushRequest, cache model.X
 			cb.credentialSocketExist = true
 		}
 	}
+	if features.MetadataExchange {
+		mxLabelsSets := sets.New[string]()
+		if features.MetadataExchangeAdditionalLabels != "" {
+			for _, label := range strings.Split(features.MetadataExchangeAdditionalLabels, ",") {
+				mxLabelsSets.Insert(label)
+			}
+		}
+		mxLabels := mxLabelsSets.UnsortedList()
+		sort.Strings(mxLabels)
+
+		cb.tcpClusterMxFilter = xdsfilters.TCPClusterMx(mxLabels...)
```

Comment:
I don't see why we need to do this here. We can do it one time in xdsfilter. Then there is no change needed here at all.

You can also make `features.MetadataExchangeAdditionalLabels` already a sorted set instead of needing to compute it each time. See InsecureKubeConfigOptions as an example.

btw you can do sets.SortedList as  well to slightly simplify.

Same applies to the listener code